docs/en/ingest-management/elastic-agent/upgrade-elastic-agent.asciidoc

```diff
@@ -56,3 +60,61 @@ sudo elastic-agent upgrade 7.10.1
 
 For more command-line options, see the help for the
 <<elastic-agent-upgrade-command,`upgrade`>> command.
+
+[discrete]
+[[upgrade-7.12-lower]]
+== Upgrade from 7.12 or lower
+
+As part of the 7.13 release, we introduce <<fleet-server,{fleet-server}>>,
+a mechanism to connect your {agent}s to {fleet}. 
+
+[WARNING]
+====
+When you upgrade to 7.13 from 7.12 or lower, please note the following:
+
+* All of your existing {agent}s will be automatically unenrolled from {fleet} and will stop sending data.
+* Before you can add your {agent}s again to {fleet}, you must have {fleet-server} running.
+* All of your policies will be migrated.
+====
+
+[discrete]
+[[fleet-prereqs-upgrade]]
+=== Prerequisite
+
+You must upgrade your deployment to 7.13. If your deployment is running in our {ess-product}[hosted {ess}]
+on {ecloud}, see {cloud}/ec-upgrade-deployment.html[Upgrade versions]. If you are running a self-managed
+deployment, see {stack-ref}/upgrading-elastic-stack.html[Upgrading the {stack}].
+
+[discrete]
+[[set-up-fleet-server]]
+=== Step 1: Set up {fleet-server}
+
+Here’s what you need to set up {fleet-server} for each deployment type:
+
+--
+include::{tab-widgets}/upgrade-fleet-widget.asciidoc[]
+--
+
+[discrete]
+[[reenroll-agent]]
+=== Step 2: Install and enroll {agent}s
+
+[IMPORTANT]
+====
+When you upgrade to 7.13 from 7.12 or lower, you may see the following error message in the {es} logs:
+
+`Authentication using apikey failed - api key has been invalidated`
+
+Before installing and enrolling your {agent}s, make sure to uninstall your previous {agent}s.
+See <<uninstall-elastic-agent>>.
+====
+
+{fleet-server} and {agent} running on {fleet-server} all run on the same version.
```

Comment:
One small comment: I don't understand this sentence. Fleet Server and Elastic Agent running on Fleet Server? Doesn't Fleet Server run on Elastic Agent?

Reply:
I agree, let me make a suggestion for this and the next line.  It touches on the question I put into email in the last hour about what Fleet Server is (a stack component) or just an implementation detail that could / should be hidden.  I think it is more part of the stack, which just happens to be running on an edge host by the Elastic Agent.

Reply:
```suggestion
From the perspective of the given host, {fleet-server} will be running with the version of the installed {agent} that started it.
```